fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
+		ino := *(*uint64)(unsafe.Pointer(&buf[0]))
+
+		if int(reclen) > len(buf) {
+			return count
+		}
+		buf = buf[reclen:]
+		if ino == 0 {
```

Comment:
I meant I don't think procfs returns entries with "ino == 0". It returns the "." and ".." but those have valid inode numbers.

```
$ ls -lai /proc/self/fd/
total 0
5470776 dr-x------. 2 alban alban  0 Apr 25 16:22 .
5470775 dr-xr-xr-x. 9 alban alban  0 Apr 25 16:22 ..
5470781 lrwx------. 1 alban alban 64 Apr 25 16:22 0 -> /dev/pts/1
5470782 lrwx------. 1 alban alban 64 Apr 25 16:22 1 -> /dev/pts/1
5470783 lrwx------. 1 alban alban 64 Apr 25 16:22 2 -> /dev/pts/1
5470784 lr-x------. 1 alban alban 64 Apr 25 16:22 3 -> /proc/29189/fd
```
The off-by-2 is done in `ReadDirCount()` with `nDotFiles := 2`. I tested manually by running Scope and looking at a few processes with `ls -l /proc/<pid>/fd/ | wc -l`.